app/scripts/views/settings/recovery_codes.js

```diff
@@ -97,6 +97,19 @@ const View = FormView.extend({
   },
 
   setInitialContext(context) {
+    // if user has no totp setup then the user should not be able to
+    // directly navigate to /settings/two_step_authentication/recovery_codes
+    const that = this;
```

Comment:
We normally use `self` variable name when referencing `this`, or use the `this` directly. I think in this case you can just use `this directly below and use use an intermediate value.